src/ctap/mod.rs

```diff
@@ -759,6 +759,33 @@ impl CtapState {
         }
     }
 
+    // This checks that either the credential contains no slot id and the expected slot id
+    // is 0, or the credential contains the expected slot id.
+    fn check_slot_id_matches(
+        &mut self,
+        credential: &PublicKeyCredentialSource,
+        slot_id: usize,
+    ) -> bool {
+        if let Some(cred_slot_id) = credential.slot_id {
+            cred_slot_id == slot_id
+        } else {
+            slot_id == 0
+        }
```

Comment:
This is a one-liner that is duplicated in credential management.

Reply:
done